### DISCUSSION TITLE
Adding simple example to docs [and possibly simplifying default usage]

### DISCUSSION THREAD

**Discussion by MatthewScholefield** [fastapi-users/fastapi-users#880] [Ideas]
With the recent updates introducing getters and classes (ie. `UserManager`), the full example seems overly complex and might deter new users getting started. I'm thinking for the health of the project it might be worthwhile to create a simple example in the docs and if the simple example looks/becomes too complex perhaps introducing tiny defaults that can be picked from to simplify the default usages.

What do you think? Would such a contribution be welcome?

**Comment by frankie567** [comment]
@MatthewScholefield I know that the initial setup may look a bit complex, but there is a lot of challenges to overcome leading to this complexity. Above all, I want to be as un-opinionated as possible, either be regarding databases, actions to take after registration/forgot-password or authentication backends.

A thing which bugs me with ready-to-use samples is that 99% of the time, people will then skip the documentation and come back to Discussions to ask questions whose answers are in the doc.

That said, if you can come with a better/quicker example to get things started, I would be happy to discuss about it and introducing it in the docs.

**Reply by MatthewScholefield** [reply]
> That said, if you can come with a better/quicker example to get things started, I would be happy to discuss about it and introducing it in the docs.

Great, I'll see if I can draft something up to illustrate what I'm thinking of.

> A thing which bugs me with ready-to-use samples is that 99% of the time, people will then skip the documentation and come back to Discussions to ask questions whose answers are in the doc.

Hmm, I'd love to discuss this but first I want to ask for a quick clarification. What do you mean by "ready-to-use examples"? Isn't that what fastapi-users already offers on the full example page in the docs?

What I'm thinking of is basically two fold:

1. Making sure there are sensible default in places where there exists some such default. To remain unopinionated or in cases where such a default may lead to an anti-pattern, there can also be a "Simple" implementation within some library subpackage that contains the most basic implementation that can be extracted when the additional complexity is needed.
2. Using these simple defaults/components, write a <10 line simple example that can be included in the frontpage of the docs.

I think having #1 is incredible valuable to all new users, no matter the experience, because it reduces the cognitive burden of understanding every component of the library and allows following the documentation with a more motivated mindset. Even if these simple starter components wouldn't even make sense in a complete workable project, they allow the user to run the code from the start and then motivates the added complexity by illustrating first what the simple component cannot do.

Let me know what you think given the added context above, and once I get some clarification on your specific concern, I'd love to share my perspective on that.

**Reply by frankie567** [reply]
> Hmm, I'd love to discuss this but first I want to ask for a quick clarification. What do you mean by "ready-to-use examples"? Isn't that what fastapi-users already offers on the full example page in the docs?

It is! And that's precisely why I'm telling this: because it's *already* happening 🙃

I understand your idea of having "ready-to-use" components; even though right now I can't really imagine how it would look like. I would be curious to see how you imagine it.
